packages/verified-fetch/src/verified-fetch.ts

```diff
@@ -452,6 +460,26 @@ export class VerifiedFetch {
     [identity.code]: this.handleRaw
   }
 
+  private async handleServerTiming<T> (name: string, description: string, fn: () => Promise<T>): Promise<T> {
```

Comment:
Can we check `this.includeServerTiming`  here to encapsulate most of the logic here and avoid the measuring and assigning of `this.serverTimingHeaders`? 

That would also allow simplifying `responseWithServerTiming`  and renaming to something a bit more generic, e.g. `augmentResponse` or `handleResponse`

Reply:
yea let me peek at this

Reply:
so, it just so happened that switching the server timing handling to handleServerTiming required me to pipe things through a lot more, so I'm moving forward with this and adding the "IPNS/DNSLink resolution (if it's an ipns url)" you previously mentioned.

I didn't originally like it in `handleServerTiming` because it required plumbing the request-level "withServerTiming" into more places, but it makes more sense for multiple reasons. 

it also prevents the extra work of wrapping the timed functions.. so it's a win. thanks for the push